### PR TITLE
[PAY-655] Refactor challenges config, fix bugs

### PR DESCRIPTION
**Pull request by piazzatron**
### Description
- Share challenge config between mobile + web where possible by moving it to @common, deleted lots o code
- Consolidates `Challenges` and `TrendingChallenges` into just `Challenges`
- Stop relying on deprecated `amounts`, instead use DN amounts
- Fix a UI issue seen by @dylanjeffers with incorrect tile sizes
- Add TS package to @common (cc @dylanjeffers)
- Default to iPhone 14 simulator

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?
- Tested on simulator and web. 
- Will have AJ  test rewards thoroughly as part of next release.

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/pages/audio-rewards-page/config.tsx [hunk 1]
```diff
@@ -83,242 +80,98 @@ const linkButtonMap: Record<LinkButtonType, LinkButtonInfo> = {
...
-    title: 'Top 5 Underground Trending',
     icon: <i className='emoji large chart-increasing' />,
-    description: 'Winners are selected every Friday at Noon PT!',
-    buttonText: 'See More',
-    id: 'trending-underground'
   }
 }
+
+
+export const getChallengeConfig = (id: ChallengeRewardID) => ({
```
Comment (dharit-tan) [inline packages/web/src/pages/audio-rewards-page/config.tsx:174]: nice!

File: packages/mobile/src/utils/challenges.tsx [hunk 2]
```diff
@@ -115,207 +45,138 @@ export type ChallengesParamList = {
...
-    description: challenges.firstPlaylistDescription,
-    shortDescription: challenges.firstPlaylistShortDescription,
-    progressLabel: challenges.firstPlaylistProgressLabel,
-    buttonInfo: {
-      label: challenges.firstPlaylistButton,
-      navigation: {
-        screen: 'favorites'
+    },
+    'first-playlist': {
+      icon: Sparkles,
```
Comment (dharit-tan) [inline packages/mobile/src/utils/challenges.tsx:134]: This is probably my b but I think we wanted to change this icon to treble-clef across the board.
https://linear.app/audius/issue/PAY-609#comment-130e98bb

File: packages/mobile/src/screens/audio-screen/Panel.tsx [hunk 3]
```diff
@@ -72,7 +75,8 @@ export const Panel = ({
   progressLabel,
   remainingLabel,
   buttonInfo,
```
Comment (dharit-tan) [inline packages/mobile/src/screens/audio-screen/Panel.tsx:77]: can we delete this arg if it's no longer used?

File: packages/web/src/pages/audio-rewards-page/config.tsx [hunk 4]
```diff
@@ -83,242 +80,98 @@ const linkButtonMap: Record<LinkButtonType, LinkButtonInfo> = {
...
     modalButtonInfo: {
       incomplete: linkButtonMap.trendingTracks,
       inProgress: linkButtonMap.trendingTracks,
       complete: linkButtonMap.trendingTracks
     }
   },
-  'mobile-install': {
-    id: 'mobile-install' as ChallengeRewardID,
-    title: 'Get the Audius Mobile App',
+  'mobile-install':{
```
Comment (dharit-tan) [inline packages/web/src/pages/audio-rewards-page/config.tsx:120]: nit: space after colon, how come linting didn't catch this?

File: packages/mobile/src/utils/challenges.tsx (hunk 2 above)
Comment (piazzatron) [reply packages/mobile/src/utils/challenges.tsx:134]: Niiiiice catch, ty

File: packages/mobile/src/screens/audio-screen/Panel.tsx (hunk 3 above)
Comment (piazzatron) [reply packages/mobile/src/screens/audio-screen/Panel.tsx:77]: good eye

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/piazz-refactor-challenges

File: packages/common/src/models/AudioRewards.ts [hunk 5]
```diff
@@ -101,23 +99,6 @@ export type FlowSessionEvent =
   | FlowSessionPassEvent
   | FlowSessionFailEvent
 
-/**
- * Needed for notifications for now as UserChallenges might not be loaded yet
- * @deprecated amounts should be pulled in directly from user challenges instead
- */
-export const amounts: Record<ChallengeRewardID, number> = {
```
Comment (rickyrombo) [inline packages/common/src/models/AudioRewards.ts:108]: <3

File: packages/common/src/utils/challenges.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    title: 'Complete Your Profile',
+    description: (challenge) =>
+      `Complete your Audius profile to earn ${challenge?.amount} $AUDIO`,
+    fullDescription: () =>
+      'Fill out the missing details on your Audius profile and start interacting with tracks and artists!',
+    progressLabel: '%0/%1 Complete',
+    panelButtonText: 'More Info'
+  },
+  'track-upload': {
+    id: 'track-upload' as ChallengeRewardID,
```
Comment (rickyrombo) [inline packages/common/src/utils/challenges.ts:91]: nit, do we need this cast?

File: packages/common/src/utils/challenges.ts [hunk 7]
```diff
@@ -0,0 +1,147 @@
+import { ChallengeRewardID, OptimisticUserChallenge } from '../models'
```
Comment (rickyrombo) [inline packages/common/src/utils/challenges.ts:1]: not gonna check this file too carefully, assuming it's the same as before but moved and git was dumb

File: packages/mobile/package.json [hunk 8]
```diff
@@ -7,7 +7,7 @@
     "android": "ENVFILE=.env.prod react-native run-android --variant=prodDebug",
     "android:stage": "ENVFILE=.env.stage react-native run-android --variant=bounceDebug --appIdSuffix bounce",
     "android:dev": "ENVFILE=.env.dev react-native run-android --variant=prodDebug",
-    "ios": "react-native run-ios",
+    "ios": "react-native run-ios --simulator \"iPhone 14 Pro\"",
```
Comment (rickyrombo) [inline packages/mobile/package.json:10]: Do we want this to be hardcoded? or should this be something configured in the dev's machine as the default instead?

File: packages/common/src/utils/challenges.ts [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+    id: 'ref-v',
+    title: 'Invite your Fans',
+    description: (challenge) =>
+      `Earn up to ${formatNumberCommas(challenge?.totalAmount ?? '')} $AUDIO`,
+    fullDescription: (challenge) =>
+      `Invite your fans! You’ll earn ${challenge?.amount} $AUDIO for each fan who joins with your link (and they’ll get an $AUDIO too)`,
+    progressLabel: '%0/%1 Invites Accepted',
+    remainingLabel: '%0/%1 Invites Remain',
+    panelButtonText: 'Invite your Fans',
+    verifiedChallenge: true
```
Comment (rickyrombo) [inline packages/common/src/utils/challenges.ts:41]: nit: booleans should follow `isXXX` or `hasXXX` naming convention per our styleguide

**Review by rickyrombo — APPROVED** [review]
HUUUGE :tada:

File: packages/common/src/utils/challenges.ts (hunk 6 above)
Comment (piazzatron) [reply packages/common/src/utils/challenges.ts:91]: we do not!

File: packages/common/src/utils/challenges.ts (hunk 9 above)
Comment (piazzatron) [reply packages/common/src/utils/challenges.ts:41]: true nice

File: packages/mobile/package.json (hunk 8 above)
Comment (piazzatron) [reply packages/mobile/package.json:10]: React Native seems to default to 13 (at least on this version), which isn't included in the latest XCode by default – I think this is a safe approach

**Review by dharit-tan — APPROVED** [review]

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/piazz-refactor-challenges

**Pull request merged**